src/browser/navigation-panel.js

```diff
@@ -64,9 +65,18 @@ define((require, exports, module) => {
         submitKey: 'Enter',
         isFocused: inputCursor.get('isFocused'),
         selection: inputCursor.get('isFocused'),
-        onFocus: _ => inputCursor.set('isFocused', true),
-        onBlur: _ => inputCursor.set('isFocused', false),
-        onChange: event => inputCursor.set('value', event.target.value),
+        onFocus: event => {
+          computeSuggestions(event.target.value, suggestionsCursor);
+          inputCursor.set('isFocused', true);
+        },
+        onBlur: event => {
+          resetSuggestions(suggestionsCursor);
+          inputCursor.set('isFocused', false)
+        },
+        onChange: event => {
+          computeSuggestions(event.target.value, suggestionsCursor);
```

Comment:
It maybe worth to `require('lang/functional').throttle(computeSuggestion)` to reduce number of computations.